lib/ansible/galaxy/collection/__init__.py

```diff
@@ -1566,6 +1569,23 @@ def _resolve_depenency_map(
         include_signatures,  # type: bool
 ):  # type: (...) -> dict[str, Candidate]
     """Return the resolved dependency map."""
+    try:
+        dist = distribution('ansible-core')
+    except PackageNotFoundError:
+        req = None
+    else:
+        req = next((rr for r in (dist.requires or []) if (rr := PkgReq(r)).name == 'resolvelib'), None)
+    finally:
+        if req is None:
+            # TODO: replace the hardcoded versions with a warning if the dist info is missing
+            # display.warning("Unable to find 'ansible-core' distribution requirements to verify the resolvelib version is supported.")
+            if RESOLVELIB_VERSION < RESOLVELIB_LOWERBOUND or RESOLVELIB_VERSION >= RESOLVELIB_UPPERBOUND:
```

Comment:
It's purely cosmetic, but this could be simplified:

```suggestion
            if not RESOLVELIB_LOWERBOUND <= RESOLVELIB_VERSION < RESOLVELIB_UPPERBOUND:
```